### ISSUE TITLE
ls --help not working on Mac

### ISSUE THREAD

**Issue by cecileane** [swcarpentry/shell-novice#447]
On line 222 of [_episodes/02-filedir.md](https://github.com/swcarpentry/shell-novice/blame/gh-pages/_episodes/02-filedir.md), this command does not work on Max OSX, which implements BSD `ls` rather than GNU `ls`:

`$ ls --help`

I get this on a Mac:

``` shell
$ ls --help
ls: illegal option -- -
usage: ls [-ABCFGHLOPRSTUWabcdefghiklmnopqrstuwx1] [file ...]
```

See [this](http://unix.stackexchange.com/questions/207136/why-does-my-ls-help-not-work) on that topic. I suggest replacing `ls --help` by `man ls`, which works for both BSD and GNU tools.

**Comment by morgantaschuk** [comment]
The docs say that `man ls` doesn't work on Git Bash though, which is the recommended terminal client on Windows.

**Comment by cecileane** [comment]
I see! so there is no one-fits-all solution. Have both `ls --help` and `man ls` plus a note, then?

**Comment by gdevenyi** [comment]
Sure, that sounds good. @cecileane can you make PR with this change?

**Comment by rgaiacs** [comment]
Can we officially suggest that learners use their favourite search engine to look for "man command"? And have a additional box saying that they can try `man command`, `command --help` or `command -h` on their terminal but if it is going to work or not will be a matter of environment?

**Comment by iglpdc** [comment]
I'd prefer to have `man command` as the "recommended" way of checking the documentation, leaving the box for Windows as is. The use case is, again, using a computer without a graphical interface such as a cluster, AWS machine, etc... 

`command -h` or `command --help` usually show a subset of the documentation, mostly the usage part, but may be used for another options. Two examples relevant to the lesson are `echo -h` or `echo --help`, which just prints out `-h` or `--help` , and `sort -h` , which understand "human" numbers.

We could add another box "Other ways of getting help" mentioning that `command -h` or `command --help`  are options that usually print a small subset of the documentation.

**Comment by morgantaschuk** [comment]
I agree that `man` is the recommended way of checking documentation, but for Windows users who are (usually) newer to the shell than people using other operating systems, this tosses them out of the lesson into "it doesn't work" land.

I found [an answer on Superuser](http://superuser.com/questions/295398/man-pages-for-git-bash-on-windows-7) that says that you can install `man` on Git Bash using `MinGW-get install msys-man`. I can't test that until I get home tonight, but perhaps it can be added to the pre-workshop setup instructions and all will be well.

**Comment by jrleeman** [comment]
This does show up in other places as well, for example the finding things has ```grep --help```, which won't work on Macs. Should a Windows/Mac/Linux recommended way be shown in all cases?

**Comment by danmichaelo** [comment]
The problem with different solutions for Win/Mac/Linux is that it makes teaching harder, so I would prefer relying on `man`. Did you try the `MinGW-get install msys-man`, @morgantaschuk ? If that works, it would be interesting to see if it could be added to the https://github.com/swcarpentry/windows-installer

**Comment by rgaiacs** [comment]
> so I would prefer relying on `man`.

`man` doesn't work all the times.

**Comment by danmichaelo** [comment]
> `man` doesn't work all the times.

Not sure if I get you. By all the times, do you mean that it doesn't work on Windows? Or that man pages is not available for some commands?

**Comment by gdevenyi** [comment]
Fixed by #813
